buildSrc/src/main/java/org/elasticsearch/gradle/JdkDownloadPlugin.java

```diff
@@ -57,218 +38,101 @@
 
     @Override
     public void apply(Project project) {
-        NamedDomainObjectContainer<Jdk> jdksContainer = project.container(
-            Jdk.class,
-            name -> new Jdk(name, project.getConfigurations().create("jdk_" + name), project.getObjects())
-        );
-        project.getExtensions().add(EXTENSION_NAME, jdksContainer);
-
-        project.afterEvaluate(p -> {
-            for (Jdk jdk : jdksContainer) {
-                jdk.finalizeValues();
+        Attribute<String> artifactType = Attribute.of("artifactType", String.class);
```

Comment:
Should we give this a more specific name?

Reply:
Its something that already is provided by gradle itself. therefore I kept the name